README.md
<p align="center">
  <img width="120" height="120" src="https://raw.githubusercontent.com/tianyong90/we-vue/master/img/logo.png">
  <h2 align="center">WeVue, 不止是 Vue.js + weui!</h2>
</p>

<p align="center">
  <a href="https://www.npmjs.com/package/we-vue"><img src="https://img.shields.io/npm/v/we-vue.svg" alt="npm"></a>
  <a href="https://circleci.com/gh/tianyong90/we-vue"><img src="https://circleci.com/gh/tianyong90/we-vue.svg?style=svg"></a>
  <a href="https://github.com/standard/standard"><img src="https://img.shields.io/badge/code_style-standard-brightgreen.svg"></a>
  <a href="https://codecov.io/github/tianyong90/we-vue?branch=master"><img src="https://img.shields.io/codecov/c/github/tianyong90/we-vue/master.svg"></a>
  <a href="https://opensource.org/licenses/MIT"><img src="https://img.shields.io/badge/License-MIT-blue.svg"></a>
  <a href="https://makeapullrequest.com"><img src="https://img.shields.io/badge/PRs-welcome-brightgreen.svg?style=flat-square"></a>
</p>

we-vue 是一套基于 Vue.js 的移动关组件库，结合 weui 官方样式库，封装了一系列组件，适合微信公众号等移动端开发。

## 特性

* 35+ 个组件
* 完善的单元测试
* 详细的在线文档和示例
* 支持 Typescript

### [在线文档（https://wevue.org）](https://wevue.org)

### [在线示例（https://demo.wevue.org）](https://demo.wevue.org)

<div align="center">
  <img src="https://raw.githubusercontent.com/tianyong90/we-vue/master/img/qrcode-demo.png" width="180">
</div>

## 安装

```shell
// yarn 推荐
$ yarn add we-vue

// npm
$ npm install we-vue -S
```

## 开发

**we-vue3.0 版本开始使用 lerna 工具，以 monorepo 模式进行开发，packages 目录下包含了 demo, docs 以及 we-vue 三个包，分别为示例，在线文档和组件库。如果有兴趣参与贡献代码，需要对 lerna 有一定的了解。除此之外，3.0使用 Typescript 对组件包进行了重构，因此也需要一定的 TS 基础。**

1. 克隆代码

```shell
// GitHub
$ git clone https://github.com/tianyong90/we-vue.git

// 码云（为方便国内用户，本项目也托管到码云，码云仅作定期同步，不保证代码为最新）
$ git clone https://gitee.com/tianyong/we-vue.git
```

2. 安装依赖

```shell
$ cd we-vue && lerna bootstrap
```

3. 运行本地服务并开发

```shell
// 开发组件及 demo
$ yarn run dev

// 开发在线文档
$ yarn run dev:docs
```

## 参与贡献

对于对本项目作出过贡献的朋友表示诚挚的感谢！同时也感谢提出过建议或反馈 Bug 的朋友。:smile:

<a href="https://github.com/tianyong90/we-vue/graphs/contributors"><img src="https://opencollective.com/we-vue/contributors.svg?width=890" /></a>


如果你有兴趣参与贡献代码或者提出建议，请阅读 [贡献指南](.github/CONTRIBUTING.md)

## 感谢

本项目从创建到现在，参考了不少流行的 vue 组件库，其中最主要的有 Mint-UI, 有赞 vant 以及 Vuetify。部分代码移植于这些 MIT 开源项目，同时又涉及大量变动，未及一一标注。在此表示感谢！

## License

MIT


CHANGELOG.md
v3

picker

column.values ==> column.options

## v2.2.5 (2018-8-27)

### 优化

- dialog 中使用 transition-group 替代 transition

---
## v2.2.4 (2018-8-27)

### 修复

- datetime-picker 边界值计算问题

---
## v2.2.3 (2018-8-27)

### 修复

- 修复 search-bar 在 iOS 中键盘无法显示“搜索”按钮的问题

---
## v2.2.1 (2018-8-25)

### 优化

- search-bar 中的输入框使用 type="search"

---
## v2.2.0 (2018-8-19)

### 新增

- 新增 Tabs 选项卡组件，将用于替代 Navbar 组件

---
## v2.1.6 (2018-8-6)

### 修复

- 修复 detetime-picker 使用自定义数值模板时可能无法正常显示初始值问题

---
## v2.1.5 (2018-7-23)

### 修复

- 修复 detetime-picker 某些场景下数值问题

---
## v2.1.4 (2018-6-19)

### 优化

- 优化 input 组件长度验证逻辑

---
## v2.1.3 (2018-6-6)

### 优化

- 调整 package.json 设置，files 包含 types 文件夹

### 修复

- 修复 infinite-scroll 事件监听在页面切换后可能无法正确移除的问题

---
## v2.1.2 (2018-5-8)

### 优化

- 完善 types

---
## v2.1.1 (2018-5-2)

### 优化

- 加入 typescript 描述文件

### 构建

- 使用 commitzen 及 conventional-changelog

---
## v2.1.0 (2018-4-6)

### 优化

- 重构 number-spinner 组件，美化样式，优化其输入框对输入、粘贴事件处理逻辑

### 构建

- 升级 webpack 到 v4,调整相关配置
- 使用 mini-css-extract-plugin 替代 extract-text-webpack-plugin
- webpack 相关配置调整

---
## v2.0.4 (2018-4-2)

### 修复

- 修复 popup 组件 lockOnScroll 无法生效的 BUG

---
## v2.0.3 (2018-3-27)

### 修复

- 修复 checklist 组件，align 属性无法正常渲染的问题

---
## v2.0.2 (2018-3-15)

### 修复

- 修正 dialog 构建方法，解决 Dialog({}) 调用时对话框不显示的问题

## v2.0.1 (2018-3-5)

### 修复

- 调整 toast 样式，解决纯文本提示时不居中的问题

### 优化

- 纯文本 toast 提示支持多行文本

---
## v2.0.0 (2018-2-12)

以下更新内容相对于 v1.6

### 代码质量控制

- 全面的单元测试，覆盖组件源码 95% 以上
- 更规范的代码，使用 eslint-plugin-vue 插件检查及修正代码
- 使用 CircleCI 自动测试，确保组件可用性

### 重要更改及优化

- 重构 swipe 组件
  - 属性 speed => duration，默认值由 300 改为 500
  - 属性 auto => autoplay，无默认值
  - 取消 condinuous 属性
  - 优化连续滑动操作体验
  
- 优化 slider 组件
  - 新增 enableClick 组件，支持点击操作（默认开启）
  - show-value-box 更名为 show-value
  - 优化属性值校验
  - 简化计算逻辑
  
- 重构 dialog 组件
  - 组件返回 Promise 对象
  - shouCancelBtn => showCancelButton
  - shouConfirmBtn => showConfirmButton
  - cancelText => cancelButtonText
  - confirmText => confirmButtonText
  - 优化动画效果

- cell-swipe 优化
  - 组件外部点击时自动复位
  
- poopup 重构
  - 事件名调整 show => open, hide => close
  - 使用 :visible.sync="visible" 形式控制 popup 是否显示
  - 支持动画效果

- toast
  - 调整 api，支持 success, fail, loading, text 等类型提示
  - loading() 方法整合原 Indicator 组件功能

- 删除 Indicator 组件，其功能整合到 Toast 组件中，使用 Toast.loading() 方法

- picker
  - 使用 :visible.sync="visible" 形式控制 picker 是否显示
  - 新增 visibleItem 属性，定义可见选项数
  - slots 属性重命名为 columns

- datetime-picker
  - 根据 picker 的变化进行相应调整

- flex-item
  - 新增 offset 属性

- preview 组件重命名为 form-preview

- search 组件
  - search 组件重命名为 search-bar
  - search-bar 组件新增 result-text-key 属性
  - search-bar 新增 `click-result` 事件，点击搜索结果项时触发

- radio
  - options 属性设为 required
  - value 支持 Object, Bookean, String 多种类型

- checklist 组件
  - 优化 change 事件触发逻辑，避免选中值实际没有改变时触发该事件

---
## v2.0.0-alpha.5 (2018-2-7)

### 新增

- flex-item 组件新增 offset 属性

---
## v2.0.0-alpha.4 (2018-2-3)

### 非兼容更改

- preview 组件重命名为 form-preview

---
## v2.0.0-alpha.3 (2018-2-1)

### 非兼容更改

- search 组件重命名为 search-bar

### 新增

- search-bar 组件新增 result-text-key 属性
- search-bar 新增 `click-result` 事件，点击搜索结果项时触发

---
## v2.0.0-alpha.2 (2018-1-29)

### 优化

- radio 组件 options 属性设为 required
- radio 组件 value 支持 Object, Bookean, String 多种类型
- 优化 checklist 组件 change 事件触发逻辑，避免选中值实际没有改变时触发该事件

---
## v2.0.0-alpha.1 (2018-1-20)

### 代码质量控制

- 全面的单元测试，覆盖组件源码 95% 以上
- 更规范的代码，使用 eslint-plugin-vue 插件检查及修正代码
- 使用 CircleCI 自动测试，确保组件可用性

### 重要更改及优化

- 重构 swipe 组件
  - 属性 speed => duration，默认值由 300 改为 500
  - 属性 auto => autoplay，无默认值
  - 取消 condinuous 属性
  - 优化连续滑动操作体验
  
- 优化 slider 组件
  - 新增 enableClick 组件，支持点击操作（默认开启）
  - show-value-box 更名为 show-value
  - 优化属性值校验
  - 简化计算逻辑
  
- 重构 dialog 组件
  - 组件返回 Promise 对象
  - shouCancelBtn => showCancelButton
  - shouConfirmBtn => showConfirmButton
  - cancelText => cancelButtonText
  - confirmText => confirmButtonText
  - 优化动画效果

- cell-swipe 优化
  - 组件外部点击时自动复位
  
- poopup 重构
  - 事件名调整 show => open, hide => close
  - 使用 :visible.sync="visible" 形式控制 popup 是否显示
  - 支持动画效果

- toast
  - 调整 api，支持 success, fail, loading, text 等类型提示
  - loading() 方法整合原 Indicator 组件功能

- 删除 Indicator 组件，其功能整合到 Toast 组件中，使用 Toast.loading() 方法

- picker
  - 使用 :visible.sync="visible" 形式控制 picker 是否显示
  - 新增 visibleItem 属性，定义可见选项数
  - slots 属性重命名为 columns

- datetime-picker
  - 根据 picker 的变化进行相应调整

---
## v1.6.2 (2018-1-8)

### 修复

- 方法名拼写错误

---
## v1.6.1 (2018-1-7)

### 修复

- 修复两处方法名拼写错误

---
## v1.6.1 (2018-1-7)

### 新特性

- input, textarea 组件支持定义 focus, blur 事件处理方法

---
## v1.6.0 (2017-12-5)

### 优化

- actionsheet, dialog 组件使用过渡动效

### 非兼容性更改
    
- cell, cell-swipe, grid-item, footer-link, media-box 等具有跳转功能的组件，新增 `url` 属性，同时  `to` 属性支持传入对象值。对于仅使用 `vue-router` 跳转的情况，此更改无影响。但对于不使用 `vue-router` 的场景下，需要使用 `url` 来指定跳转目标链接。 

---
## v1.5.3 (2017-11-28)

### 修复

- input 组件中一处拼写错误
- 修正 textarea 组件初始值可能超出 max-length 的问题

---
## v1.5.2 (2017-11-27)

### 优化

- 重构 switch, cell-swipe, picker 滑动操作相关代码
- 解除 css3transfom 依赖，减小文件体积

---
## v1.5.1 (2017-11-17)

### 修复

- 紧急修正 v1.5.0 中 Toast, Indicator, TopTips 组件失效问题

---
## v1.5.0 (2017-11-16)

- picker 组件优化，增加滚动效果，并可点击选择 [#18](https://github.com/tianyong90/we-vue/pull/18)

### 其它调整

- 放弃 cooking 构建工具，回归 vue-cli，优化开发体验
- 不再构建 commonjs（仅构建 umd）
- 部分加入单元测试（后期将进一步覆盖）
- 源码格式规范化

### 非兼容性更改

- lib 目录下不再有 we-vue.common.js 及 style.min.js， 若单独手动引用，使用 `lin/index.js` 和 `lib/style.css`

---
## v1.4.18 (2017-10-31)

### 修复

- 修复 datetime-pick 一处可能引起死循环的 bug

---
## v1.4.17 (2017-10-24)

- popup 组件 heigt 属性，支持指定 % 及 vh 形式高度
- switch 禁用状态对应颜色调整
- input 组件新增 maxlength, name, minlength, autofocus, auto-complete 属性

---
## v1.4.16 (2017-10-24)

- switch 组件支持滑动操作，同时兼容点击操作

---
## v1.4.15 (2017-10-23)

### 新增

- input 组件字段验证模式支持输入时验证 (即 onInput 模式)，默认启用

---
## v1.4.14 (2017-09-23)

### 修复

- navbar 指示条宽度高处 bug

---
## v1.4.13 (2017-09-23)

### 新增

- navbar 的指示条支持动效，新增 animate 属性

### change

- navbar 高度调整
- navbar 指示条默认宽度调整，由 3px 改为 2px

### 修复

- navbar 禁用项点事件处理逻辑优化
- navbar 禁用项文字颜色设置

---
## v1.4.12 (2017-09-14)

### 新增

- 新增 number-spinner 组件

### 修复

lib/icon.js
/******/ (function(modules) { // webpackBootstrap
/******/ 	// The module cache
/******/ 	var installedModules = {};

/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {

/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId])
/******/ 			return installedModules[moduleId].exports;

/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			exports: {},
/******/ 			id: moduleId,
/******/ 			loaded: false
/******/ 		};

/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);

/******/ 		// Flag the module as loaded
/******/ 		module.loaded = true;

/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}


/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;

/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;

/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = "/";

/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(0);
/******/ })
/************************************************************************/
/******/ ([
/* 0 */
/***/ function(module, exports, __webpack_require__) {

	'use strict';

	Object.defineProperty(exports, "__esModule", {
	  value: true
	});

	var _icon = __webpack_require__(52);

	Object.defineProperty(exports, 'default', {
	  enumerable: true,
	  get: function get() {
	    return _interopRequireDefault(_icon).default;
	  }
	});

	function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/***/ },
/* 1 */,
/* 2 */,
/* 3 */,
/* 4 */,
/* 5 */,
/* 6 */,
/* 7 */,
/* 8 */,
/* 9 */,
/* 10 */
/***/ function(module, exports, __webpack_require__) {

	"use strict";

	exports.__esModule = true;

	var _defineProperty = __webpack_require__(11);

	var _defineProperty2 = _interopRequireDefault(_defineProperty);

	function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

	exports.default = function (obj, key, value) {
	  if (key in obj) {
	    (0, _defineProperty2.default)(obj, key, {
	      value: value,
	      enumerable: true,
	      configurable: true,
	      writable: true
	    });
	  } else {
	    obj[key] = value;
	  }

	  return obj;
	};

/***/ },
/* 11 */
/***/ function(module, exports, __webpack_require__) {

	module.exports = { "default": __webpack_require__(12), __esModule: true };

/***/ },
/* 12 */
/***/ function(module, exports, __webpack_require__) {

	__webpack_require__(13);
	var $Object = __webpack_require__(16).Object;
	module.exports = function defineProperty(it, key, desc){
	  return $Object.defineProperty(it, key, desc);
	};

/***/ },
/* 13 */
/***/ function(module, exports, __webpack_require__) {

	var $export = __webpack_require__(14);
	// 19.1.2.4 / 15.2.3.6 Object.defineProperty(O, P, Attributes)
	$export($export.S + $export.F * !__webpack_require__(24), 'Object', {defineProperty: __webpack_require__(20).f});

/***/ },
/* 14 */
/***/ function(module, exports, __webpack_require__) {

	var global    = __webpack_require__(15)
	  , core      = __webpack_require__(16)
	  , ctx       = __webpack_require__(17)
	  , hide      = __webpack_require__(19)
	  , PROTOTYPE = 'prototype';

	var $export = function(type, name, source){
	  var IS_FORCED = type & $export.F
	    , IS_GLOBAL = type & $export.G
	    , IS_STATIC = type & $export.S
	    , IS_PROTO  = type & $export.P
	    , IS_BIND   = type & $export.B
	    , IS_WRAP   = type & $export.W
	    , exports   = IS_GLOBAL ? core : core[name] || (core[name] = {})
	    , expProto  = exports[PROTOTYPE]
	    , target    = IS_GLOBAL ? global : IS_STATIC ? global[name] : (global[name] || {})[PROTOTYPE]
	    , key, own, out;
	  if(IS_GLOBAL)source = name;
	  for(key in source){
	    // contains in native
	    own = !IS_FORCED && target && target[key] !== undefined;
	    if(own && key in exports)continue;
	    // export native or passed
	    out = own ? target[key] : source[key];
	    // prevent global pollution for namespaces
	    exports[key] = IS_GLOBAL && typeof target[key] != 'function' ? source[key]
	    // bind timers to global for call from export context
	    : IS_BIND && own ? ctx(out, global)
	    // wrap global constructors for prevent change them in library
	    : IS_WRAP && target[key] == out ? (function(C){
	      var F = function(a, b, c){
	        if(this instanceof C){
	          switch(arguments.length){
	            case 0: return new C;
	            case 1: return new C(a);
	            case 2: return new C(a, b);
	          } return new C(a, b, c);
	        } return C.apply(this, arguments);
	      };
	      F[PROTOTYPE] = C[PROTOTYPE];
	      return F;
	    // make static versions for prototype methods
	    })(out) : IS_PROTO && typeof out == 'function' ? ctx(Function.call, out) : out;
	    // export proto methods to core.%CONSTRUCTOR%.methods.%NAME%
	    if(IS_PROTO){
	      (exports.virtual || (exports.virtual = {}))[key] = out;
	      // export proto methods to core.%CONSTRUCTOR%.prototype.%NAME%
	      if(type & $export.R && expProto && !expProto[key])hide(expProto, key, out);
	    }
	  }
	};
	// type bitmap
	$export.F = 1;   // forced
	$export.G = 2;   // global
	$export.S = 4;   // static
	$export.P = 8;   // proto
	$export.B = 16;  // bind
	$export.W = 32;  // wrap
	$export.U = 64;  // safe
	$export.R = 128; // real proto method for `library` 
	module.exports = $export;

/***/ },
/* 15 */
/***/ function(module, exports) {

	// https://github.com/zloirock/core-js/issues/86#issuecomment-115759028
	var global = module.exports = typeof window != 'undefined' && window.Math == Math
	  ? window : typeof self != 'undefined' && self.Math == Math ? self : Function('return this')();
	if(typeof __g == 'number')__g = global; // eslint-disable-line no-undef

/***/ },
/* 16 */
/***/ function(module, exports) {

	var core = module.exports = {version: '2.4.0'};
	if(typeof __e == 'number')__e = core; // eslint-disable-line no-undef

/***/ },
/* 17 */
/***/ function(module, exports, __webpack_require__) {

	// optional / simple context binding
	var aFunction = __webpack_require__(18);
	module.exports = function(fn, that, length){
	  aFunction(fn);
	  if(that === undefined)return fn;
	  switch(length){
	    case 1: return function(a){
	      return fn.call(that, a);
	    };
	    case 2: return function(a, b){
	      return fn.call(that, a, b);
	    };
	    case 3: return function(a, b, c){
	      return fn.call(that, a, b, c);
	    };
	  }
	  return function(/* ...args */){
	    return fn.apply(that, arguments);
	  };
	};

/***/ },
/* 18 */
/***/ function(module, exports) {

	module.exports = function(it){
	  if(typeof it != 'function')throw TypeError(it + ' is not a function!');
	  return it;
	};

/***/ },
/* 19 */
/***/ function(module, exports, __webpack_require__) {

	var dP         = __webpack_require__(20)
	  , createDesc = __webpack_require__(28);
	module.exports = __webpack_require__(24) ? function(object, key, value){
	  return dP.f(object, key, createDesc(1, value));
	} : function(object, key, value){
	  object[key] = value;
	  return object;
	};

/***/ },
/* 20 */
/***/ function(module, exports, __webpack_require__) {

	var anObject       = __webpack_require__(21)
	  , IE8_DOM_DEFINE = __webpack_require__(23)
	  , toPrimitive    = __webpack_require__(27)
	  , dP             = Object.defineProperty;

	exports.f = __webpack_require__(24) ? Object.defineProperty : function defineProperty(O, P, Attributes){
	  anObject(O);
	  P = toPrimitive(P, true);
	  anObject(Attributes);
	  if(IE8_DOM_DEFINE)try {
	    return dP(O, P, Attributes);
	  } catch(e){ /* empty */ }
	  if('get' in Attributes || 'set' in Attributes)throw TypeError('Accessors not supported!');
	  if('value' in Attributes)O[P] = Attributes.value;
	  return O;
	};

/***/ },
/* 21 */
/***/ function(module, exports, __webpack_require__) {

	var isObject = __webpack_require__(22);
	module.exports = function(it){
	  if(!isObject(it))throw TypeError(it + ' is not an object!');
	  return it;
	};

/***/ },
/* 22 */
/***/ function(module, exports) {

	module.exports = function(it){
	  return typeof it === 'object' ? it !== null : typeof it === 'function';
	};

/***/ },
/* 23 */
/***/ function(module, exports, __webpack_require__) {

	module.exports = !__webpack_require__(24) && !__webpack_require__(25)(function(){
	  return Object.defineProperty(__webpack_require__(26)('div'), 'a', {get: function(){ return 7; }}).a != 7;
	});

/***/ },
/* 24 */
/***/ function(module, exports, __webpack_require__) {

	// Thank's IE8 for his funny defineProperty
	module.exports = !__webpack_require__(25)(function(){
	  return Object.defineProperty({}, 'a', {get: function(){ return 7; }}).a != 7;
	});

/***/ },
/* 25 */
/***/ function(module, exports) {

	module.exports = function(exec){
	  try {
	    return !!exec();
	  } catch(e){
	    return true;
	  }
	};

/***/ },
/* 26 */
/***/ function(module, exports, __webpack_require__) {

	var isObject = __webpack_require__(22)
	  , document = __webpack_require__(15).document
	  // in old IE typeof document.createElement is 'object'
	  , is = isObject(document) && isObject(document.createElement);
	module.exports = function(it){
	  return is ? document.createElement(it) : {};
	};

/***/ },
/* 27 */
/***/ function(module, exports, __webpack_require__) {

	// 7.1.1 ToPrimitive(input [, PreferredType])
	var isObject = __webpack_require__(22);
	// instead of the ES6 spec version, we didn't implement @@toPrimitive case
	// and the second argument - flag - preferred type is a string
	module.exports = function(it, S){
	  if(!isObject(it))return it;
	  var fn, val;
	  if(S && typeof (fn = it.toString) == 'function' && !isObject(val = fn.call(it)))return val;
	  if(typeof (fn = it.valueOf) == 'function' && !isObject(val = fn.call(it)))return val;
	  if(!S && typeof (fn = it.toString) == 'function' && !isObject(val = fn.call(it)))return val;
	  throw TypeError("Can't convert object to primitive value");
	};

/***/ },
/* 28 */
/***/ function(module, exports) {

	module.exports = function(bitmap, value){
	  return {
	    enumerable  : !(bitmap & 1),
	    configurable: !(bitmap & 2),
	    writable    : !(bitmap & 4),
	    value       : value
	  };
	};

/***/ },
/* 29 */,
/* 30 */,
/* 31 */,
/* 32 */,
/* 33 */,
/* 34 */,
/* 35 */,
/* 36 */,
/* 37 */,
/* 38 */,
/* 39 */,
/* 40 */,
/* 41 */,
/* 42 */,
/* 43 */,
/* 44 */,
/* 45 */,
/* 46 */,
/* 47 */,
/* 48 */,
/* 49 */,
/* 50 */,
/* 51 */,
/* 52 */
/***/ function(module, exports, __webpack_require__) {

	var __vue_exports__, __vue_options__
	var __vue_styles__ = {}

	/* script */
	__vue_exports__ = __webpack_require__(53)

	/* template */
	var __vue_template__ = __webpack_require__(54)
	__vue_options__ = __vue_exports__ = __vue_exports__ || {}
	if (
	  typeof __vue_exports__.default === "object" ||
	  typeof __vue_exports__.default === "function"
	) {
	__vue_options__ = __vue_exports__ = __vue_exports__.default
	}
	if (typeof __vue_options__ === "function") {
	  __vue_options__ = __vue_options__.options
	}

	__vue_options__.render = __vue_template__.render
	__vue_options__.staticRenderFns = __vue_template__.staticRenderFns

	module.exports = __vue_exports__


/***/ },
/* 53 */
/***/ function(module, exports, __webpack_require__) {

	'use strict';

	Object.defineProperty(exports, "__esModule", {
	  value: true
	});

	var _defineProperty2 = __webpack_require__(10);

	var _defineProperty3 = _interopRequireDefault(_defineProperty2);

	__webpack_require__(!(function webpackMissingModule() { var e = new Error("Cannot find module \"weui/dist/style/weui.min.css\""); e.code = 'MODULE_NOT_FOUND'; throw e; }()));

	function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

	exports.default = {
	  name: 'wv-icon',

	  props: {
	    type: {
	      type: String,
	      required: true
	    },
	    large: Boolean
	  },

	  computed: {
	    classObject: function classObject() {
	      var _ref;

	      var classType = 'weui-icon-' + this.type;

	      return _ref = {}, (0, _defineProperty3.default)(_ref, classType, true), (0, _defineProperty3.default)(_ref, 'weui-icon_msg', this.large), _ref;
	    }
	  }
	};

/***/ },
/* 54 */
/***/ function(module, exports) {

	module.exports={render:function (){var _vm=this;var _h=_vm.$createElement;
	  return _vm._c('i', {
	    class: _vm.classObject
	  })
	},staticRenderFns: []}

/***/ }
/******/ ]);